mPower2/mPower2/View Controllers/StudyBurstViewController.swift

```diff
@@ -173,12 +170,61 @@ class StudyBurstViewController: UIViewController {
         }
     }
     
-    func showCompletionView() {
+    func showCompletionTask() {
         // TODO: syoung 06/20/2018 Update title/text to "congradulations" with party sprinkles.
         self.navFooterView.nextButton?.setTitle(Localization.buttonDone(), for: .normal)
         
-        // Call through to the today view to show the completion tasks.
-        (self.navigationController?.viewControllers.first as? TodayViewController)?.showStudyBurstCompletionView()
+        // If there is a task to do today, then push it.
+        if let taskPath = studyBurstManager.completionTaskPath() {
+            let vc = RSDTaskViewController(taskPath: taskPath)
+            vc.delegate = self
+            self.present(vc, animated: true, completion: nil)
+        }
+    }
+}
+
+extension StudyBurstViewController: RSDTaskViewControllerDelegate {
+    
+    open func taskController(_ taskController: RSDTaskController, didFinishWith reason: RSDTaskFinishReason, error: Error?) {
+        // dismiss the view controller
+        if let vc = taskController as? UIViewController {
+            if vc == self.navigationController?.topViewController {
+                self.navigationController?.popToRootViewController(animated: true)
+            }
+            else {
+                self.dismiss(animated: true) {
+                    self.navigationController?.popToRootViewController(animated: true)
+                }
+            }
+        }
+        // Let the schedule manager handle the cleanup.
+        studyBurstManager.taskController(taskController, didFinishWith: reason, error: error)
+    }
+
+    func taskController(_ taskController: RSDTaskController, readyToSave taskPath: RSDTaskPath) {
+        studyBurstManager.taskController(taskController, readyToSave: taskPath)
+    }
+
+    func taskController(_ taskController: RSDTaskController, asyncActionControllerFor configuration: RSDAsyncActionConfiguration) -> RSDAsyncActionController? {
+        return studyBurstManager.taskController(taskController, asyncActionControllerFor:configuration)
+    }
+    
+    func taskViewController(_ taskViewController: UIViewController, viewControllerFor step: Any) -> UIViewController? {
```

Comment:
WARNING: Code stomp. Please revert.

The `RSDTaskViewControllerDelegate` methods have been moved to the `TodayViewController` because the UI design requires different ordering of what views to display based on a whole series of "what it" scenarios.

The `TodayViewController` is **always** the root view controller, but under certain circumstances, this view controller is **not** shown next. Sometimes the navigation will push on surveys in front of this view. That's why I changed the "back" button to pop to root.

Please move this logic to the `TodayViewController` method. Thanks!